File: Src/java/cql-to-elm/src/main/java/org/cqframework/cql/cql2elm/model/invocation/CombineInvocation.java
package org.cqframework.cql.cql2elm.model.invocation;

import static java.util.Objects.requireNonNull;

import java.util.ArrayList;
import java.util.List;
import org.hl7.elm.r1.Combine;
import org.hl7.elm.r1.Expression;

public class CombineInvocation extends OperatorExpressionInvocation<Combine> {
    public CombineInvocation(Combine expression) {
        super(expression);
    }

    @Override
    public List<Expression> getOperands() {
        Combine combine = expression;
        ArrayList<Expression> ops = new ArrayList<>();
        ops.add(combine.getSource());
        if (combine.getSeparator() != null) {
            ops.add(combine.getSeparator());
        }
        return ops;
    }

    @Override
    public void setOperands(List<Expression> operands) {
        requireNonNull(operands, "operands cannot be null.");
        require(!operands.isEmpty() && operands.size() <= 2, "Combine operator requires one or two operands.");

        expression.setSource(operands.get(0));
        if (operands.size() > 1) {
            expression.setSeparator(operands.get(1));
        }
    }
}

File: Src/java/cql-to-elm/src/main/java/org/cqframework/cql/cql2elm/model/invocation/CombineInvocation.kt
package org.cqframework.cql.cql2elm.model.invocation

import java.util.*
import org.hl7.elm.r1.Combine
import org.hl7.elm.r1.Expression

class CombineInvocation(expression: Combine) : OperatorExpressionInvocation<Combine>(expression) {
    override var operands: List<Expression>
        get() = listOfNotNull(expression.source, expression.separator)
        set(operands) {
            require(operands.size in 1..2) { "Combine operator requires one or two operands." }

            expression.source = operands[0]
            if (operands.size > 1) {
                expression.separator = operands[1]
            }
        }
}
